packages/ceramic-docid/src/index.ts

```diff
@@ -0,0 +1,195 @@
+import CID from 'cids'
+const multibase = require('multibase')
+const subnets = require('./subnet-table.json')
+const doctypes = require('./doctype-table.json')
+const varint = require('varint')
+const uint8ArrayConcat = require('uint8arrays/concat')
+const uint8ArrayToString = require('uint8arrays/to-string')
+const uint8ArrayEquals = require('uint8arrays/equals')
+const DOCID_CODEC = 206
+
+// Definition
+// '<multibase-prefix><multicodec-docid><subnet><doctype><genesis-cid-bytes>'
+
+class DocId {
+  /**
+   * Create a new DocId.
+   *
+   * @param {string|Uint8Array}  doctype
+   * @param {CID|string}         [cid]
+   * @param {string}             [multihash = 'devnet']
+   * @param {string}             [multibaseName = 'base32']
+   *
+   * @example
+   * new DocId(<docType>, <CID>|<cidStr>)
+   * new DocId(<docType>, <CID>|<cidStr>, <subnet>, <multibaseName>)
+   * new DocId(<docIdStr>)
+   * new DocId(<bytes>, <multibaseName>)  
+   */
+
+  private _docType: string 
+  private _subnet: string
+  private _cid: CID
+  private _cidBytes: Uint8Array
+  private _multibaseName: string
+
+  constructor (doctype: string | Uint8Array, cid: CID | string, subnet?: string, multibaseName?: string) {
+    if (doctype instanceof Uint8Array) {
+      // Given uint8 array DocId
+      const bytes = doctype 
+      this._multibaseName = typeof cid === 'string' ? cid : 'base32'
+      this._initFromBytes(bytes)
+    } else if (Object.keys(doctypes).includes(doctype)) {
+      // From args
+      if (!doctype) throw new Error('constructor: doctype required')
+      if (!cid) throw new Error('constructor: cid required')
+      this._docType = doctype
+      this._subnet = subnet || 'devnet'
+      this._multibaseName = multibaseName || 'base32'
+      this._cid = (typeof cid === 'string') ? new CID(cid) : cid
+      this._cidBytes = this._cid.bytes
+    } else if (typeof doctype === 'string') {
+      // From DocIdString
+      const docIdStr = doctype
+      this._multibaseName = multibase.isEncoded(docIdStr)
+      const bytes = multibase.decode(docIdStr).slice(2)
+      this._initFromBytes(bytes)
+    } else {
+      throw new Error('constructor: Invalid Arguments or CID')
+    }
+  }
```

Comment:
Can we change the constructor to be less ambiguous? 
My preference would be:
* Have a static `fromBytes` function
* Have a static `fromString` function
* doctype is either the string name or the number (this allows ppl to use custom doctypes without adding to table